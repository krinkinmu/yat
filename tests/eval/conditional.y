def main(variable) {
    if (variable) {
        def foo() {
            print variable;
        }
    } else {
        def foo() {
            read variable;
        }
    }
    foo;
}

read arg;
main(arg)();

read other;
main(other)();

read variable;
main(variable)();

if (1) {
} else {
  print(-1);
}

if (0) {
  print(-1);
}

if (0) {
  print(-1);
} else {
}

if (main(10)()) {
} else {
}

if (main(20)()) {
  print(30);
} else {
}

if (main(40)()) {
} else {
  print(50);
}

if (main(60)()) {
  print(70);
} else {
  print(80);
}
