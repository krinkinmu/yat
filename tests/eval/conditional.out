1313
1313
1313
30
1313
1313
70
